applications/teletype/src/templates/teletype_customer_update.erl

```diff
@@ -0,0 +1,157 @@
+                                          ]).
+
+-spec handle_req(wh_json:object(), wh_proplist()) -> 'ok'.
+handle_req(JObj, _Props) ->
+    'true' = wapi_notifications:customer_update_v(JObj),
+    DataJObj = wh_json:normalize(JObj),
+    AccountId = wh_json:get_value(<<"account_id">>, DataJObj),
+    case teletype_util:is_notice_enabled(AccountId, JObj, ?TEMPLATE_ID) of
+        'false' -> lager:debug("notification handling not configured for this account");
+        'true' -> process_req(DataJObj)
+    end.
+
+-spec process_req(wh_json:object()) -> 'ok'.
+process_req(DataJObj) ->
+    SenderId = wh_json:get_value(<<"account_id">>, DataJObj),
+    ViewOpts = [{'startkey', [SenderId]}
+               ,{'endkey', [SenderId, wh_json:new()]}
+               ],
+    case couch_mgr:get_results(?WH_ACCOUNTS_DB, <<"accounts/listing_by_children">>, ViewOpts) of
+        {'ok', Accounts} ->
+            process_accounts(Accounts, DataJObj);
+        {'error', _Reason} = E ->
+            lager:info("failed to load children. error: ~p", [E]),
+            E
+    end.
+
+process_accounts(Accounts, DataJObj) ->
+    case wh_json:get_value(<<"recipient_id">>, DataJObj) of
+        'undefined' ->
+            [process_account(wh_json:get_value(<<"value">>, Account), DataJObj) || Account <- Accounts];
+        RecipientId ->
+            [process_account(wh_json:get_value(<<"value">>, Account), DataJObj) || Account <- Accounts, wh_json:get_value(<<"id">>, Account) == RecipientId]
+    end.
+
+process_account(AccountJObj, DataJObj) ->
+    AccountId = wh_json:get_value(<<"id">>, AccountJObj),
+    AccountDb = wh_util:format_account_id(AccountId, 'encoded'),
+    case couch_mgr:get_results(AccountDb,<<"users/crossbar_listing">>,[]) of
+        {'ok', Users} ->
+            Templates = teletype_templates:fetch(?TEMPLATE_ID, DataJObj),
+            select_users_to_update(lists:map(fun(User) -> wh_json:get_value(<<"value">>, User) end, Users), DataJObj, Templates);
+        {'error', _Reason} = E ->
+            lager:info("failed to load users from: ~p", [AccountDb]),
+            E
+    end.
+
+select_users_to_update(Users, DataJObj, Templates) ->
+    case wh_json:get_value(<<"user_type">>, DataJObj) of
+        <<UserId:32/binary>> ->
+            [send_update_to_user(User, DataJObj, Templates) || User <- Users, wh_json:get_value(<<"id">>, User) == UserId];
+        <<"all_users">> ->
+            [send_update_to_user(User, DataJObj, Templates) || User <- Users];
+        _ ->
+            [send_update_to_user(User, DataJObj, Templates) || User <- Users, wh_json:get_value(<<"priv_level">>, User) == <<"admin">>]
+    end.
+
+send_update_to_user(UserJObj, DataJObj, Templates) ->
+    Macros = [{<<"system">>, teletype_util:system_params()}
+              ,{<<"system">>, teletype_util:account_params(DataJObj)}
+              | build_macro_data(UserJObj)
+             ],
+    RenderedTemplates = [{ContentType, teletype_util:render(?TEMPLATE_ID, Template, Macros)}
```

Comment:
How you render has changed; please refer to a module like `teletype_low_balance` for how to render your templates now. Quickly,

```erlang
    RenderedTemplates = teletype_templates:render(?TEMPLATE_ID, Macros, DataJObj),
    {'ok', TemplateMetaJObj} = teletype_templates:fetch_notification(?TEMPLATE_ID, teletype_util:find_account_id(DataJObj)),
```

This should be easier for everyone. It will also bypass the render farm when using the system's template, speeding up processing.